### PR TITLE
1. citybutton now is 0.5 transparent to makesure it don't cover the near unit. 2. add "replace image with emoji" setting, some image can be replaced with the emoji image which be created from the system font( only show the setting when it runs on android and version>=8).  3. now great scientist does the same as original game. 4. delete freetype because it is not used. 5. now all files  are written in kotlin and game runs well in android and desktop.

### PR DESCRIPTION
**Pull request by lishaoxia1985**

### CODE REVIEW

**Comment by yairm210** [comment]
A. If something isn't usable on MacOS then I don't want it in the game, I just managed to get all the Desktop standalones working
B. I assume you checked the city button and it's readable even though it's semi transparent?

**Comment by lishaoxia1985** [comment]
A. i am not sure, the different system support different emoji version, it will be good in android 9 but bad in android 7.  Unless we decide to unuse emoji in game,  if not giving a choise to user is a good choice. it is up to you.
B. yes, becuase  i have give everycity  button a clear civ icon, it should be 100% tranparent, the city button should only have a civoutcolor edge, but i do not know how to do.

**Comment by lishaoxia1985** [comment]
@yairm210 now it can be only used when android 8 and above.

File: android/src/com/unciv/app/AndroidLauncher.java [hunk 1]
```diff
@@ -16,6 +16,7 @@ protected void onCreate (Bundle savedInstanceState) {
 		String version = BuildConfig.VERSION_NAME;
 
 		config.useImmersiveMode=true;
+		//config.useGL30 = true;
```
Comment (lishaoxia1985) [inline android/src/com/unciv/app/AndroidLauncher.java]: it have some bugs, forbidding is good choice.

**Comment by yairm210** [comment]
What bugs does it have in GL 3.0?

**Comment by lishaoxia1985** [comment]
it runs very slowly, you can see in the game.

**Comment by lishaoxia1985** [comment]
@yairm210 now great scientist will be the same as orginal game.

File: core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt [hunk 2]
```diff
@@ -197,11 +195,11 @@ class UnitActions {
 
 
         if (unit.name == "Great Scientist" && !unit.isEmbarked()) {
-            actionList += UnitAction("Discover Technology", unit.currentMovement >0
+            actionList += UnitAction("Hurry Research", unit.civInfo.tech.currentTechnologyName() != null
```
Comment (lishaoxia1985) [inline core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt]: in original game, when currentTechnologyName()=null, the science great scientist get will add to overflow science directly, i think it is a bug.

**Pull request closed without merging**